### PR TITLE
Fixed setting the source to an empty string

### LINKED ISSUE #1785: Setting src to an empty string throws an error.
Is there a good way to stop the current video from loading? `player.src('')` works but spews errors into the console, which is undesirable.

Why not dispose of the player? I want to continue playing the same video later, I just don't want it eating bandwidth while "paused".

### PR DESCRIPTION
**Pull request by heff**
Fixes #1785

### CODE REVIEW

File: test/unit/media.js [hunk 1]
```diff
@@ -229,3 +229,26 @@ test('should add the source hanlder interface to a tech', function(){
...
+    trigger: this.noop
+  };
+
+  // Define a new tech class
+  var Tech = videojs.MediaTechController.extend();
+  // Extend Tech with source handlers
+  vjs.MediaTechController.withSourceHandlers(Tech);
+
+  Tech.nativeSourceHandler = {
+    handleSource: function(){ usedNative = true; }
```
Comment (mmcc) [inline test/unit/media.js]: I don't love the weird indirection here. Why not just do the assertion right in this callback and set the number of expected assertions to 1?

**Comment by mmcc** [comment]
Otherwise :thumbsup:

**Comment by heff** [comment]
Cleaned up the test order

File: test/unit/media.js [hunk 2]
```diff
@@ -229,3 +229,25 @@ test('should add the source hanlder interface to a tech', function(){
   tech.dispose();
   ok(disposeCalled, 'the handler dispose method was called when the tech was disposed');
 });
+
+test('should handle unsupported sources with the source hanlder API', function(){
```
Comment (gkatsev) [inline test/unit/media.js]: handler*

**Pull request closed without merging**